packages/block-editor/src/components/inspector-controls-tabs/use-inspector-controls-tabs.js

```diff
@@ -54,5 +76,12 @@ export default function useInspectorControlsTabs( blockName ) {
 		tabs.push( TAB_SETTINGS );
 	}
 
-	return tabs;
+	const tabSettings = useSelect( ( select ) => {
+		return select( blockEditorStore ).getSettings()
+			.__experimentalBlockInspectorTabs;
+	}, [] );
+
+	const showTabs = getShowTabs( blockName, tabSettings );
+
+	return showTabs ? tabs : [];
```

Comment:
Shall we return a const `EMPTY_ARRAY` from outside the hook scope? That way we have a stable reference for the hook to avoid unnecessary changes.